pkg/dr/health.go

```diff
@@ -0,0 +1,81 @@
+package dr
+
+import (
+	ctx "context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	rookv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type secretData struct {
+	Key     string `json:"key"`
+	MonHost string `json:"mon_host"`
+}
+
+func Health(context *k8sutil.Context, operatorNamespace, cephClusterNamespace string, args []string, ExitOnError bool) {
+	fmt.Println("INFO: fetching the cephblockpools with mirroring enabled")
+	blockPoolList, err := context.RookClientset.CephV1().CephBlockPools(cephClusterNamespace).List(ctx.TODO(), v1.ListOptions{})
+	if err != nil {
+		log.Error(err)
+	}
+
+	var mirrorBlockPool rookv1.CephBlockPool
+	for _, blockPool := range blockPoolList.Items {
+		if blockPool.Spec.Mirroring.Enabled {
+			mirrorBlockPool = blockPool
+			fmt.Printf("found %s cephblockpool with mirroring enabled\n", mirrorBlockPool.Name)
+		}
+	}
+
+	if mirrorBlockPool.Name == "" {
+		fmt.Printf("DR is not confiqured, cephblockpool with mirroring enabled not found.\n")
+	}
+
+	secretNameList := mirrorBlockPool.Spec.Mirroring.Peers.SecretNames
+	secretName := secretNameList[0]
+	secretName = strings.TrimSuffix(secretName, "]")
+	secretName = strings.TrimPrefix(secretName, "[")
+
+	secret, err := context.Clientset.CoreV1().Secrets(cephClusterNamespace).Get(ctx.TODO(), secretName, v1.GetOptions{})
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Instead of making these fatal, how about only failing this section of the output? For example, factor out lines 42-76 to a separate method where the secret must be parsed. Then even if the secret fails to load, the status on lines 78-80 can still continue.

Reply:
we want the secrets details, otherwise ceph status will loaded from local cluster and not from pee cluster.

Reply:
but refactored based on secret extraction